### PR TITLE
ORCH-416 Add support of ScannerForMSBuild >= 4.1.0.1148

### PR DESCRIPTION
**Pull request by m-g-sonar**

### CODE REVIEW

File: sonar-orchestrator/src/main/java/com/sonar/orchestrator/build/ScannerForMSBuildExecutor.java [hunk 1]
```diff
@@ -53,8 +58,20 @@ BuildResult execute(ScannerForMSBuild build, Configuration config, Map<String, S
     }
   }
 
-  private static Command createCommand(ScannerForMSBuild build, Map<String, String> adjustedProperties, File runnerScript) {
-    Command command = Command.create(runnerScript.getAbsolutePath());
+  private static Command createCommand(ScannerForMSBuild build, Configuration config, Map<String, String> adjustedProperties, File runnerScript) {
```
Comment (SonarTech) [inline sonar-orchestrator/src/main/java/com/sonar/orchestrator/build/ScannerForMSBuildExecutor.java]: Code Smell Code Smell: Remove this unused method parameter "config". ([squid:S1172](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1172))

[See it in SonarQube](https://next.sonarqube.com/sonarqube/project/issues?id=org.sonarsource.orchestrator%3Aorchestrator-parent&pullRequest=45&issues=AWMruZAtau6S9RpGg6Z_&open=AWMruZAtau6S9RpGg6Z_)

[ISSUE_KEY]: AWMruZAtau6S9RpGg6Z_

File: sonar-orchestrator/src/main/java/com/sonar/orchestrator/build/ScannerForMSBuildInstaller.java [hunk 2]
```diff
@@ -70,35 +76,56 @@ private File install(Location location, File toDir, boolean useOldScript) {
...
+        // change permissions on binary files from sonar-scanner included in scanner
+        setSonarScannerBinariesAsExecutable(scannerDir);
+      }
     } catch (Exception e) {
       throw new IllegalStateException("Fail to unzip Scanner for MSBuild from " + zipFile + " to " + toDir, e);
     }
   }
 
-  private File locateZip(Version scannerVersion) {
+  private void setSonarScannerBinariesAsExecutable(File scannerDir) {
```
Comment (SonarTech) [inline sonar-orchestrator/src/main/java/com/sonar/orchestrator/build/ScannerForMSBuildInstaller.java]: Code Smell Code Smell: Make "setSonarScannerBinariesAsExecutable" a "static" method. ([squid:S2325](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS2325))

[See it in SonarQube](https://next.sonarqube.com/sonarqube/project/issues?id=org.sonarsource.orchestrator%3Aorchestrator-parent&pullRequest=45&issues=AWMruZBBau6S9RpGg6aA&open=AWMruZBBau6S9RpGg6aA)

[ISSUE_KEY]: AWMruZBBau6S9RpGg6aA

File: sonar-orchestrator/src/main/java/com/sonar/orchestrator/build/ScannerForMSBuild.java [hunk 3]
```diff
@@ -80,6 +81,12 @@ public ScannerForMSBuild setUseDotNetCore(boolean useDotnetCore) {
     return this;
   }
 
+  public ScannerForMSBuild setDotNetCoreExecutable(File dotNetCoreExecutable) {
+    this.dotNetCoreExecutable = dotNetCoreExecutable;
+    // automatically enable usage of .Net Core
```
[the remark this answers is not in the document]
Comment (m-g-sonar) [reply sonar-orchestrator/src/main/java/com/sonar/orchestrator/build/ScannerForMSBuild.java]: I don't see why I should not flag the boolean telling that .NET Core should be used here. I know that having an executable should be enough to tell that we need it, but I prefer to have it coherent.

[the remark this answers is not in the document]
Comment (simonbrandhof) [reply sonar-orchestrator/src/main/java/com/sonar/orchestrator/build/ScannerForMSBuild.java]: that was not my point. My sentence was not clear. Calling `setUseDotNetCore(true)` is useless _for clients_. Calling `setDotNetCoreExecutable()` is enough and that's good. It just deserves to be documented.

**Pull request merged**